core/server/proxy/src/main/java/alluxio/proxy/s3/S3RestServiceHandler.java

```diff
@@ -1115,6 +1267,7 @@ private void deleteExistObject(final FileSystem fs, AlluxioURI objectURI, Boolea
   }
 
   private String parsePath(String bucketPath, String prefix, String delimiter) throws S3Exception {
+    // TODO(czhu): allow non-"/" delimiters
```

Comment:
With your changes, any delimiter should work right? If the delimiter is / then don't do a recursive listing, otherwise for all other delimiters, do a recursive listing, and it should work?

Reply:
That would work, but you _could_ do a "smart" traversal so that you don't need a recursive listing. That's why I left it as a TODO.

eg: with the following file structure
```
/test/delim/
-> foo
-> dir/
     -> img1.png
-> img_store/
     -> img2.png
```

Then you're expected to return the following common prefixes:
```
% aws s3api list-objects-v2 --bucket=czhu-dev --prefix=test/delim --delimiter=/img 
{
    "Contents": [
        {
            "Key": "test/delim/foo",
            "LastModified": "2022-04-20T23:41:19+00:00",
            "ETag": "\"911df44b7ff57801ca8d74568e4ebfbe\"",
            "Size": 27040,
            "StorageClass": "STANDARD"
        }
    ],
    "CommonPrefixes": [
        {
            "Prefix": "test/delim/dir/img"
        },
        {
            "Prefix": "test/delim/img"
        }
    ]
}
```

So ideally you could do a depth-first-traversal on each path and prematurely return whenever you match the delimiter. Maybe _THAT_ should be the TODO as an improvement over naively doing a recursive listing though? Wdyt @yuzhu @tcrain?

Reply:
Actually I don't think there's enough time to verify non-"/" delimiters at this point anyway, but yes you're correct that the functionality _is_ actually there if we just did a recursive listing.